lhotse/recipes/csj.py

```diff
@@ -0,0 +1,246 @@
+        manifests = read_manifests_if_cached(
+            dataset_parts=dataset_parts, output_dir=output_dir, types=["cuts"],
+        )
+    
+    with ThreadPoolExecutor(num_jobs) as ex:
+        for part in tqdm(dataset_parts, desc="Dataset parts"):
+            logging.info(f"Processing CSJ subset: {part}")
+            if manifests_exist(types=["cuts"], part=part, output_dir=output_dir):
+                logging.info(f"CSJ subset: {part} already prepared - skipping.")
+                continue
+            
+            recordings = []
+            supervisions = []
+            part_path = transcript_dir / part
+            futures = []
+            
+            for wavlist in part_path.glob("*/*-wav.list"):
+                template = wavlist.as_posix().rstrip('-wav.list')
+                spk = wavlist.name.rstrip('-wav.list')
+                futures.append(
+                    ex.submit(parse_one_recording, template, wavlist, spk)
+                )
+
+            for future in tqdm(futures, desc="Processing", leave=False):
+                result = future.result()
+                assert result
+                recording, segments = result
+                recordings.append(recording)
+                supervisions.extend(segments)
+            
+            recording_set = RecordingSet.from_recordings(recordings)
+            supervision_set = SupervisionSet.from_segments(supervisions)
+            validate_recordings_and_supervisions(recording_set, supervision_set)
+            
+            if output_dir:
+                supervision_set.to_file(output_dir / f"supervisions_{part}.jsonl.gz")
+                recording_set.to_file(output_dir / f"recordings_{part}.jsonl.gz")
+            
+    
+    # Create train and valid cuts
+    logging.info(f"Loading, trimming, and shuffling the remaining core+noncore cuts.")
+    recording_set = RecordingSet.from_file(output_dir / "recordings_core.jsonl.gz") \
+        + RecordingSet.from_file(output_dir / "recordings_noncore.jsonl.gz")
+    supervision_set = SupervisionSet.from_file(output_dir / "supervisions_core.jsonl.gz") \
+        + SupervisionSet.from_file(output_dir / "supervisions_noncore.jsonl.gz")
+        
+    cut_set = CutSet.from_manifests(
+        recordings=recording_set,
+        supervisions=supervision_set
+    )
+    cut_set = cut_set.trim_to_supervisions(keep_overlapping=False)
+    
+    cut_set = cut_set.shuffle(random.Random(RNG_SEED))
+    
+    logging.info(f"Creating valid cuts, split at {split}")
+    valid_set = CutSet.from_cuts(islice(cut_set, 0, split))
+    valid_set.to_jsonl(output_dir / "cuts_valid.jsonl.gz")
+    
+    logging.info(f"Creating train cuts")
+    train_set = CutSet.from_cuts(islice(cut_set, split, None))
+    
+    train_set = (
+        train_set
+        + train_set.perturb_speed(0.9)
+        + train_set.perturb_speed(1.1)
+    )
+    train_set.to_jsonl(output_dir / "cuts_train.jsonl.gz")
```

Comment:
Use `to_file` instead of `to_jsonl`.

Reply:
After the latest commit, this part was removed to return `SupervisionSet` and `RecordingSet` directly.